src/toggle-switch/toggle-switch.scss

```diff
@@ -2,194 +2,212 @@
 // See LICENSE.md in the project root for license terms and full copyright notice.
 @import '../style/index';
 
+$toggle-switch-handle: $iui-icons-default;
+
 @mixin iui-toggle-switch {
   @include iui-reset;
   display: flex;
   align-items: center;
+  gap: $iui-s;
+  align-items: center;
   font-size: $iui-font-size;
   width: fit-content;
-  padding: ($iui-baseline/2) 0;
   user-select: none;
   cursor: pointer;
   position: relative;
+  isolation: isolate;
   @include themed {
     color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-2));
   }
 
-  > input {
-    width: 0;
-    height: 0;
+  // Disabled
+  &.iui-disabled {
+    cursor: not-allowed;
+    @include themed {
+      color: t(iui-text-color-muted);
+    }
+  }
+
+  // IE solution
+  // For labels on the right
+  > .iui-label {
+    margin-right: $iui-s;
+    @supports (gap: $iui-s) {
+      margin-right: 0;
+    }
+  }
+
+  // For labels on the left
+  .iui-toggle ~ .iui-label {
+    margin-left: $iui-s;
+    margin-right: 0;
+    @supports (gap: $iui-s) {
+      margin-left: 0;
+    }
+  }
+}
+
+@mixin iui-toggle {
+  display: grid;
+  grid-template: 'toggle';
+  align-items: center;
+
+  > input[type='checkbox'] {
+    @include iui-reset;
+    @include iui-focus($thickness: 1px, $offset: 1px);
+    grid-area: toggle;
+    display: flex;
+    align-items: center;
+    position: relative;
+    cursor: pointer;
+    width: ($toggle-switch-handle + $iui-xs) * 2;
+    height: auto;
+    padding: 1px;
+    border-radius: 9999px;
     appearance: none;
-    opacity: 0;
-    position: absolute;
+    z-index: 1;
+
+    &::before {
```

Comment:
What is the benefit having toggle switch body inside `:before`? You wrote it claened up CSS, but not it has even more lines of code 😀

Reply:
We did this because we thought the 1px border was the cause of the off-centering issue in checkbox / radio buttons.  It seems like that may not be the case, so my latest push puts all of the body styling (background color & border) on the `<input>` instead of in `::before`.